105-strimzi-drain-cleaner-namespace-filtering-proposal.md

```diff
@@ -0,0 +1,269 @@
+5. If watched: Continue with existing drain cleaner logic
+
+## Affected/not affected projects
+
+**Affected Projects:**
+- **strimzi-drain-cleaner**: Core implementation of namespace filtering functionality, including packaging and deployment templates
+
+**Not Affected Projects:**
+- **strimzi-operator**: No changes required to main operator logic
+- **strimzi-bridge**: No impact on bridge functionality
+- **strimzi-canary**: Independent component, no changes needed
+- **test-container**: No changes to testing infrastructure
+
+## Implementation Details
+
+### Code Changes
+
+**Core Logic** (`src/main/java/io/strimzi/ValidatingWebhook.java`):
+- Add `@ConfigProperty` for `strimzi.drain.watch.namespaces`
+- Implement `isNamespaceWatched()` method
+- Add namespace check before pod retrieval
+
+**Configuration** (`src/main/resources/application.properties`):
+```properties
+# Comma-separated list of namespaces to watch for eviction events. Empty means all namespaces.
+strimzi.drain.watch.namespaces=
+```
+
+**Deployment Templates** (in `packaging/` directory only):
+- `packaging/install/*/060-Deployment.yaml`
+- `packaging/helm-charts/helm3/strimzi-drain-cleaner/values.yaml`
+
+### Behavior Examples
+
+**Example 1: Watched Namespace**
+```yaml
+env:
+  - name: STRIMZI_DRAIN_WATCH_NAMESPACES
+    value: "kafka-prod,kafka-dev"
+```
+- Eviction request for pod in `kafka-prod` → Full drain cleaner processing
+- Result: Normal Strimzi drain cleaner behavior
+
+**Example 2: Unwatched Namespace**
+- Eviction request for pod in `app-namespace` → Immediate allow
+- Log: `DEBUG: Ignoring eviction request for Pod some-pod in namespace app-namespace - namespace not in watch list`
+- Result: Eviction proceeds normally, no API calls made
+
+**Example 3: Default Behavior**
+```yaml
+env:
+  - name: STRIMZI_DRAIN_WATCH_NAMESPACES
+    value: ""
+```
+- All namespaces processed (current behavior)
+- Full backward compatibility maintained
+
+### Security Considerations
+
+1. **No RBAC Changes Required**: Existing cluster-wide permissions remain necessary for webhook operation
+2. **Application-Level Security**: Filtering prevents unnecessary API calls to restricted namespaces
+3. **Audit Trail**: All filtering decisions logged at appropriate levels
+4. **Fail-Safe Design**: Invalid configuration defaults to watching all namespaces
+
+## Alternatives Considered
+
+### 1. RBAC-Level Filtering
+**Approach**: Use Kubernetes RBAC to restrict access
+**Rejected Because**: 
+- Webhooks require cluster-wide registration
+- Cannot selectively grant namespace access for webhook operations
+- Would break fundamental webhook architecture
+
+### 2. Multiple Drain Cleaner Instances
+**Approach**: Deploy separate drain cleaners per namespace
+**Rejected Because**:
+- Requires multiple ValidatingWebhookConfigurations
+- Potential conflicts and performance issues
+- Goes against "one per cluster" design principle
+- Complex certificate and configuration management
```

Comment:
You cannot really reject it. Your proposal opens the doors to this and users will find out about. So as mentioned above, you need to include this as an unwanted side effect of this proposal.